ntp/4.2.8p15/ntp-4.2.8p15.patch

```diff
@@ -0,0 +1,533 @@
++	AC_ARG_ENABLE(
++	    [autokey],
++	    AS_HELP_STRING(
++		[--enable-autokey],
++		[+ support NTP Autokey protocol]
++		),
++	    [ntp_autokey=$enableval],
++	    [ntp_autokey=$ntp_openssl]
++	)
++	case "$ntp_autokey" in
++	 no)
++	    ;;
++	 *)
++	    case "$ntp_openssl" in
++	     no)
++		AC_MSG_WARN([Disabling Autokey, --enable-autokey requires --with-crypto.])
++		ntp_autokey=no
++		;;
++	     *)
++		AC_DEFINE([AUTOKEY], [1], [Support NTP Autokey protocol?])
++		ntp_autokey=yes
++		;;
++	    esac
++	    ;;
++	esac
++	AC_MSG_RESULT([$ntp_autokey])
++fi
+ 
+ AC_SUBST([MAKE_CHECK_LAYOUT])
+ AC_MSG_CHECKING([if we want to run check-layout])
+diff --git a/include/libssl_compat.h b/include/libssl_compat.h
+index 2a3697c..2964515 100644
+--- a/include/libssl_compat.h
++++ b/include/libssl_compat.h
+@@ -20,6 +20,8 @@
+ #ifndef NTP_LIBSSL_COMPAT_H
+ #define NTP_LIBSSL_COMPAT_H
+ 
++#ifndef WITH_WOLFSSL
++
+ #include "openssl/evp.h"
+ #include "openssl/dsa.h"
+ #include "openssl/rsa.h"
+@@ -116,4 +118,6 @@ extern int	sslshim_X509_get_signature_nid(const X509 *x);
+ #endif /* OPENSSL_VERSION_NUMBER < v1.1.0 */
+ /* ----------------------------------------------------------------- */
+ 
++#endif /* !WITH_WOLFSSL */
++
+ #endif /* NTP_LIBSSL_COMPAT_H */
+diff --git a/include/ntp_crypto.h b/include/ntp_crypto.h
+index 32134a8..6b3af65 100644
+--- a/include/ntp_crypto.h
++++ b/include/ntp_crypto.h
+@@ -19,11 +19,15 @@
+ #define	CRYPTO_CONF_NID   9	/* specify digest name */
+ 
+ #ifdef AUTOKEY
+-#ifndef OPENSSL
++#if !defined(OPENSSL)
+ #error AUTOKEY should be defined only if OPENSSL is.
+ invalidsyntax: AUTOKEY should be defined only if OPENSSL is.
+ #endif
+ 
++#ifdef WITH_WOLFSSL
++#include "wolfssl/options.h"
++#endif
++
+ #include "openssl/bn.h"
+ #include "openssl/evp.h"
+ #include "ntp_calendar.h"	/* for fields in the cert_info structure */
+diff --git a/include/ntp_md5.h b/include/ntp_md5.h
+index 06c90b2..258c936 100644
+--- a/include/ntp_md5.h
++++ b/include/ntp_md5.h
+@@ -7,14 +7,17 @@
+ #define NTP_MD5_H
+ 
+ #ifdef OPENSSL
++#ifdef WITH_WOLFSSL
++# include <wolfssl/options.h>
++#endif /* WITH_WOLFSSL */
+ # include <openssl/evp.h>
+ # include "libssl_compat.h"
+-# ifdef HAVE_OPENSSL_CMAC_H
++# if defined(HAVE_OPENSSL_CMAC_H) || (defined(WITH_WOLFSSL) && defined(WOLFSSL_CMAC))
+ #  include <openssl/cmac.h>
+ #  define CMAC                  "AES128CMAC"
+ #  define AES_128_KEY_SIZE      16
+-# endif /*HAVE_OPENSSL_CMAC_H*/
+-#else	/* !OPENSSL follows */
++# endif /* HAVE_OPENSSL_CMAC_H || (WITH_WOLFSSL && WOLFSSL_CMAC) */
++#else /* !OPENSSL follows */
+ /*
+  * Provide OpenSSL-alike MD5 API if we're not using OpenSSL
+  */
+@@ -22,28 +25,28 @@
+ #  include <md5.h>
+ # else
+ #  include "isc/md5.h"
+-   typedef isc_md5_t		MD5_CTX;
+-#  define MD5Init(c)		isc_md5_init(c)
+-#  define MD5Update(c, p, s)	isc_md5_update(c, p, s)
+-#  define MD5Final(d, c)	isc_md5_final((c), (d))	/* swapped */
++   typedef isc_md5_t    MD5_CTX;
```

Comment:
This block of changes seems to be spacing only.  If we put these back, we might benefit from a shorter patch file?

Reply:
Good catch. Will remove.